src/adapters/bridge-api.ts

```diff
@@ -0,0 +1,74 @@
+import axios from "axios";
+import { z } from "zod";
+
+import { StrictSchema } from "src/utils/type-safety";
+import * as domain from "src/domain";
+
+interface Bridge {
+  token_addr: string;
+  amount: string;
+  dest_net: number;
+  dest_addr: string;
+  deposit_cnt: string;
+}
+
+interface GetBridgesParams {
+  env: domain.Env;
+  ethereumAddress: string;
+}
+
+interface GetBridgesResponse {
+  deposits?: Bridge[];
+}
+
+const bridgeParser = StrictSchema<Bridge>()(
+  z.object({
+    token_addr: z.string(),
+    amount: z.string(),
+    dest_net: z.number(),
+    dest_addr: z.string(),
+    deposit_cnt: z.string(),
+  })
+);
+
+const getBridgesResponseParser = StrictSchema<GetBridgesResponse>()(
+  z.object({
+    deposits: z.optional(z.array(bridgeParser)),
```

Comment:
In the wallet, we agreed on the policy of not failing to decode a list when a single element of the list does not meet the interface. Instead, we agreed to omit the offender. I see that this is no longer honored here. Is this decision conscious?

Reply:
Yes, this API is different. This is a readonly API and they read the data directly from the smart contract, so if one element of the list is wrong, most likely the entire list is going to be wrong. So I don't see the need of having unknown parsers, reducers etc to filter wrong elements of the list in this case.